cmd/kflex/init/init.go

```diff
@@ -81,53 +71,27 @@ func Init(ctx context.Context, kubeconfig, version, buildDate string, domain, ex
 	wg.Wait()
 }
 
-func ensureSystemDB(ctx context.Context, isOCP bool) {
-	vars := []string{
-		"primary.extendedConfiguration=max_connections=1000",
-		"primary.priorityClassName=system-node-critical",
-	}
-	if isOCP {
-		vars = append(vars,
-			"primary.podSecurityContext.fsGroup=null",
-			"primary.podSecurityContext.seccompProfile.type=RuntimeDefault",
-			"primary.containerSecurityContext.runAsUser=null",
-			"primary.containerSecurityContext.allowPrivilegeEscalation=false",
-			"primary.containerSecurityContext.runAsNonRoot=true",
-			"primary.containerSecurityContext.seccompProfile.type=RuntimeDefault",
-			"volumePermissions.enabled=false",
-			"shmVolume.enabled=false",
-		)
-	}
-	h := &helm.HelmHandler{
-		URL:         PostgresURL,
-		RepoName:    PostgresRepoName,
-		ChartName:   PostgresChartName,
-		Namespace:   util.SystemNamespace,
-		ReleaseName: PostgresReleaseName,
-		Args:        map[string]string{"set": strings.Join(vars, ",")},
-	}
-	err := helm.Init(ctx, h)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error initializing helm: %v\n", err)
-		os.Exit(1)
-	}
-
-	if !h.IsDeployed() {
-		err := h.Install()
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error installing chart: %v\n", err)
-			os.Exit(1)
-		}
-	}
-}
-
 func ensureKFlexOperator(ctx context.Context, fullVersion, domain, externalPort string, isOCP bool) {
 	version := util.ParseVersionNumber(fullVersion)
 	vars := []string{
 		fmt.Sprintf("version=%s", version),
 		fmt.Sprintf("domain=%s", domain),
 		fmt.Sprintf("externalPort=%s", externalPort),
 		fmt.Sprintf("isOpenShift=%s", strconv.FormatBool(isOCP)),
+		"postgresql.primary.extendedConfiguration=max_connections=1000",
+		"postgresql.primary.priorityClassName=system-node-critical",
+	}
+	if isOCP {
+		vars = append(vars,
+			"postgresql.primary.podSecurityContext.fsGroup=null",
+			"postgresql.primary.podSecurityContext.seccompProfile.type=RuntimeDefault",
+			"postgresql.primary.containerSecurityContext.runAsUser=null",
+			"postgresql.primary.containerSecurityContext.allowPrivilegeEscalation=false",
+			"postgresql.primary.containerSecurityContext.runAsNonRoot=true",
+			"postgresql.primary.containerSecurityContext.seccompProfile.type=RuntimeDefault",
+			"postgresql.volumePermissions.enabled=false",
+			"postgresql.shmVolume.enabled=false",
+		)
```

Comment:
I am not completely sure that the parameters for OCP are passed correctly with the new setup. I did a quick test building the chart and pushing to a temp personal repo on quay.io, building the CLI from the PR branch and running `kflex init` vs the OCP cluster, and the postgres DB does not start - stateful set reports all these issues:

```
Warning  FailedCreate      35s (x16 over 118s)  statefulset-controller  create Pod kubeflex-operator-postgresql-0 in StatefulSet kubeflex-operator-postgresql failed error: pods "kubeflex-operator-postgresql-0" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, provider "pipelines-scc": Forbidden: not usable by user or serviceaccount, provider restricted-v2: .spec.securityContext.fsGroup: Invalid value: []int64{1001}: 1001 is not an allowed group, provider restricted-v2: .containers[0].runAsUser: Invalid value: 1001: must be in the ranges: [1000730000, 1000739999], provider "restricted": Forbidden: not usable by user or serviceaccount, provider "ibm-restricted-scc": Forbidden: not usable by user or serviceaccount, provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "ibm-anyuid-scc": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "log-collector-scc": Forbidden: not usable by user or serviceaccount, provider "ibm-anyuid-hostpath-scc": Forbidden: not usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "ibm-anyuid-hostaccess-scc": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "ibm-privileged-scc": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]
```
I suspect that the sub chart ran again into that issue we saw a while ago about not being able to set values to null. Looking at the generated stateful set for postgres, it has :

```
securityContext:
        fsGroup: 1001
        seccompProfile:
          type: RuntimeDefault
```

But  for OCP fsGroup should be null: `postgresql.primary.podSecurityContext.fsGroup=null` 

Also, container has:

```
 securityContext:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
              - ALL
            runAsGroup: 0
            runAsNonRoot: true
            runAsUser: 1001
            seccompProfile:
              type: RuntimeDefault
 ```
 
 But    it should have `"postgresql.primary.containerSecurityContext.runAsUser=null"`

Reply:
Only after manually editing the stateful set to delete those two lines (fsGroup: 1001 and runAsUser: 1001), the stateful set started correctly in OCP. So we need to find a solution for nulling values in the sub chart to make this work.